Sources/SwiftLexicalLookup/LookupName.swift

```diff
@@ -12,31 +12,87 @@
 
 import SwiftSyntax
 
+@_spi(Experimental) public enum LookupImplicitNameKind {
+  /// `self` keyword representing object instance.
+  case `self`(SyntaxProtocol)
```

Comment:
It had to be `SyntaxProtocol` since we didn't have a way to represent an identifier for `self` keyword in closure capture. Now, as `self` in closure capture is just an explicit `identifier` name kind, I changed it from [`SyntaxProtocol` to `DeclSyntaxProtocol `](https://github.com/swiftlang/swift-syntax/blob/252d3be80c010add2bac9f9712ac32d9892ea574/Sources/SwiftLexicalLookup/LookupName.swift#L20) as in `.Self` case.